d2game/d2player/game_controls.go

```diff
@@ -632,106 +588,111 @@ func (g *GameControls) OnMouseButtonDown(event d2interface.MouseEvent) bool {
 	return false
 }
 
-func (g *GameControls) toggleHeroStatsPanel() {
+func (g *GameControls) clearLeftScreenSide() {
+	g.heroStatsPanel.Close()
+	g.questLog.Close()
+	g.hud.skillSelectMenu.ClosePanels()
+	g.hud.miniPanel.SetMovedRight(false)
+	g.updateLayout()
+}
+
+func (g *GameControls) clearRightScreenSide() {
+	g.inventory.Close()
+	g.skilltree.Close()
+	g.hud.skillSelectMenu.ClosePanels()
+	g.hud.miniPanel.SetMovedLeft(false)
+	g.updateLayout()
+}
+
+func (g *GameControls) clearScreen() {
+	g.clearRightScreenSide()
+	g.clearLeftScreenSide()
+	g.hud.skillSelectMenu.ClosePanels()
+	g.HelpOverlay.Close()
+}
+
+func (g *GameControls) openLeftPanel(panel Panel) {
 	if !g.HelpOverlay.IsOpen() {
-		g.hud.skillSelectMenu.LeftPanel.Close()
-		g.hud.skillSelectMenu.RightPanel.Close()
-		g.questLog.Close()
-		g.heroStatsPanel.Toggle()
-		g.hud.miniPanel.SetMovedRight(g.heroStatsPanel.IsOpen())
-		g.updateLayout()
+		isOpen := panel.IsOpen()
+
+		g.clearLeftScreenSide()
+
+		if !isOpen {
+			panel.Open()
+			g.hud.miniPanel.SetMovedRight(true)
+			g.updateLayout()
+		}
 	}
 }
 
+func (g *GameControls) openRightPanel(panel Panel) {
+	if !g.HelpOverlay.IsOpen() {
+		isOpen := panel.IsOpen()
+
+		g.clearRightScreenSide()
+
+		if !isOpen {
+			panel.Open()
+			g.hud.miniPanel.SetMovedLeft(true)
+			g.updateLayout()
+		}
+	}
+}
+
+func (g *GameControls) toggleHeroStatsPanel() {
+	g.openLeftPanel(g.heroStatsPanel)
+}
+
 func (g *GameControls) onCloseHeroStatsPanel() {
-	g.hud.miniPanel.SetMovedRight(g.heroStatsPanel.IsOpen())
-	g.updateLayout()
 }
 
 func (g *GameControls) toggleLeftSkillPanel() {
 	if !g.HelpOverlay.IsOpen() {
-		g.inventory.Close()
-		g.skilltree.Close()
-		g.questLog.Close()
-		g.heroStatsPanel.Close()
+		g.clearScreen()
 		g.hud.skillSelectMenu.ToggleLeftPanel()
 	}
 }
 
 func (g *GameControls) toggleRightSkillPanel() {
 	if !g.HelpOverlay.IsOpen() {
-		g.inventory.Close()
-		g.skilltree.Close()
-		g.questLog.Close()
-		g.heroStatsPanel.Close()
+		g.clearScreen()
 		g.hud.skillSelectMenu.ToggleRightPanel()
 	}
 }
 
 func (g *GameControls) toggleQuestLog() {
-	if !g.HelpOverlay.IsOpen() {
-		g.heroStatsPanel.Close()
-		g.hud.skillSelectMenu.LeftPanel.Close()
-		g.hud.skillSelectMenu.RightPanel.Close()
-		g.questLog.Toggle()
-		g.hud.miniPanel.SetMovedRight(g.questLog.IsOpen())
-		g.updateLayout()
-	}
+	g.openLeftPanel(g.questLog)
 }
 
 func (g *GameControls) onCloseQuestLog() {
-	g.hud.miniPanel.SetMovedRight(g.questLog.IsOpen())
-	g.updateLayout()
 }
```

Comment:
should I leave it here?

Reply:
I tested quickly, and it looks like you would also need to add a check for if the onCloseCb is nil if you remove these empty functions. `Close()` doesn't check if the function pointer is nil before trying to run it.

I think it might save someone some work in the future to leave these empty functions here, in case they need to use them, but I also can't think of a situation where these would be needed, so maybe it's best for them just to go away.